code/modules/mob/mob.dm

```diff
@@ -124,11 +124,13 @@
 			continue
 		if (!M.client || istype(M, /mob/abstract/new_player))
 			continue
-		if(get_turf(M) in messageturfs)
+		if((get_turf(M) in messageturfs) || (isobserver(M) && (M.client.prefs.toggles & CHAT_GHOSTEARS)))
```

Comment:
It might get a bit too noisy in the chat if you enable that, so it would probs be good to tie that to a different preference.

Reply:
Maybe `CHAT_GHOSTSIGHT` is better alternative?